jina/drivers/evaluate.py

```diff
@@ -118,4 +119,57 @@ def _apply_all(self,
                 gt_content = pb2array(gt_content)
             evaluation.value = self.exec_fn(doc_content, gt_content)
             evaluation.op_name = f'{self.metric}-{self.exec.metric}'
-            evaluation.ref_id = groundtruth.id
\ No newline at end of file
+            evaluation.ref_id = groundtruth.id
+
+
+class GroundTruthKVIndexDriver(BaseIndexDriver):
+    """Serialize the documents/chunks in the request to key-value JSON pairs and write it using the executor
+    """
+    def __init__(self, id_tag: str = 'id', *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._traversal_paths = ('r', )
+        self.id_tag = id_tag
```

Comment:
the motivation of using `tags[id]` instead of `.id` directly is you worry about `hash` gives them different id?

Reply:
the motivation is that is hard for the `user` to have exact control on what `id` a specific document will receive, (plus he may evaluate with only some part of the document) (he may have stored just the embedding and not other parts). 

So I imagine the user may have a dataset such as:
ID, Ranking of IDs
1, [3, 5, 6

Reply:
i feel there is different understanding on that, to me `actual` and `desired` are always eye-to-eye structural-alike. For example, 

```
actual:
    [doc.id: 1, matches: [doc.id: 45, doc.id 67, doc.id: 10],
    [doc.id: 3, matches: [doc.id: 32, doc.id 24, doc.id: 23]
    ...]
```

Then groundtruth should follow the same structure and *one only need to query for [1, 3]*:

```
desired:
    [doc.id: 1, matches: [doc.id: 45, doc.id 88, doc.id: 99],
    [doc.id: 3, matches: [doc.id: 32, doc.id 99, doc.id: 77]
    ...]
```

Reply:
My idea was about allowing the `executor` not to know anything about the `document structure` so it can play with only basic types. And since he has no control on `ids` (maybe he just wants to index in the groundtruth his own `id` (but not the actual content))

Reply:
> My idea was about allowing the executor not to know anything about the document structure so it can play with only basic types.

but this does not contradict the idea

Reply:
so if we have `doc.id` it will be a `hash` that the `user` will have almost no control about (he will need to match exactly the same document).

(In many cases the groundtruth will come in form of user-related format and would be really hard for the user to convert into a document which could have the exact `hash-id`)

Reply:
but hash-id can be changed, it is not immutable.

Reply:
Ok will revert this part and expect the user to provide an `id` they can control

Reply:
Done!